src/main/java/org/openrewrite/java/testing/junit5/MigrateFromHamcrest.java

```diff
@@ -0,0 +1,375 @@
+                    maybeRemoveImport("org.hamcrest.MatcherAssert.assertThat");
+                    String targetAssertion = getTranslatedAssert(matcherInvocation, false);
+                    if (targetAssertion.equals("")) {
+                        return mi;
+                    }
+
+                    JavaTemplate template = JavaTemplate.builder(getTemplateForMatcher(matcherInvocation,null, false))
+                      .javaParser(JavaParser.fromJavaVersion().classpathFromResources(executionContext, "junit-jupiter-api-5.9"))
+                      .staticImports("org.junit.jupiter.api.Assertions." + targetAssertion)
+                      .build();
+
+                    maybeAddImport("org.junit.jupiter.api.Assertions", targetAssertion);
+                    return template.apply(getCursor(), method.getCoordinates().replace(),
+                        getArgumentsForTemplate(matcherInvocation, null, mi.getArguments().get(0), false));
+                }
+                else throw new IllegalArgumentException("Parameter mismatch for " + mi + ".");
+            }
+            System.out.println("FINISH RUN");
+            return mi;
+        }
+
+        private String getTranslatedAssert(J.MethodInvocation methodInvocation, boolean negated) {
+            //to be replaced with a static map
+            switch (methodInvocation.getSimpleName()) {
+                case "equalTo":
+                case "emptyArray":
+                case "equalToIgnoringCase":
+                case "hasEntry":
+                case "hasSize":
+                case "hasToString":
+                    return negated ? "assertNotEquals" : "assertEquals";
+                case "closeTo":
+                case "containsString":
+                case "empty":
+                case "emptyCollectionOf":
+                case "emptyIterable":
+                case "emptyIterableOf":
+                case "endsWith":
+                case "greaterThan":
+                case "greaterThanOrEqualTo":
+                case "hasKey":
+                case "hasValue":
+                case "lessThan":
+                case "lessThanOrEqualTo":
+                case "sameInstance":
+                case "startsWith":
+                case "theInstance":
+                case "isCompatibleWith":
+                    return negated ? "assertFalse" : "assertTrue";
+                case "instanceOf":
+                case "isA":
+                    return negated ? "assertFalse" : "assertInstanceOf";
+                case "is":
+                    if (methodInvocation.getArguments().get(0).toString().startsWith("org.hamcrest")) {
+                        if (methodInvocation.getArguments().get(0) instanceof J.MethodInvocation) {
+                            return getTranslatedAssert((J.MethodInvocation)methodInvocation.getArguments().get(0), negated);
+                        } else {
+                            throw new IllegalArgumentException();
```

Comment:
Instead of throwing an exception we prefer not to make a change; would that be possible here?
Otherwise we trip up for instance a complete Spring Boot 3.1 migration, if that somewhere down to line includes the Hamcrest migration as part of the JUnit 4 to 5 migration.